questions/shop.md

```diff
@@ -0,0 +1,165 @@
+After that we are going to give the netherite ingot to the player with the tag and we will clear the diamonds from it.
+To avoid the player from a loop we will remove the tag as we dont need it more.
+
+#### Pre-1.20.5
+
+    # Command blocks
+    tag @p add buyer.netherite
+    execute as @a[tag=buyer.netherite] store result score @s diamonds run clear @s diamond 0
+    give @a[tag=buyer.netherite,scores={diamonds=5..}] netherite_ingot 1
+    clear @a[tag=buyer.netherite,scores={diamonds=5..}] diamond 5
+    tellraw @a[tag=buyer.netherite,scores={diamonds=5..}] {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+    tellraw @a[tag=buyer.netherite,scores={diamonds=..4}] {"text":"You don't have 5 diamonds","color":"dark_red"}
+    tag @a remove buyer.netherite
+
+Or if you prefer a function:
+
+    # function example:load
+    scoreboard objectives add diamonds dummy
+    
+    # function example:buy/netherite
+    execute store result score @s diamonds run clear @s diamond 0
+    execute if entity @s[scores={diamonds=5..}] run function example:buy/netherite/success
+    execute unless entity @s[scores={diamonds=5..}] run tellraw @s {"text":"You don't have 5 diamonds","color":"dark_red"}
+
+    # function example:buy/netheirte/success
+    give @s netherite_ingot 1
+    clear @s diamond 5
+    tellraw @s {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+
+> [!NOTE]
+> The function `example:buy/netherite` must be run `as` the player
+
+#### 1.20.5 and above
+
+    # Command blocks
+    tag @p add buyer.netherite
+    execute as @p[tag=buyer.netherite] store result score @s diamonds if items entity @s container.* diamond
+    give @a[tag=buyer.netherite,scores={diamonds=5..}] netherite_ingot 1
+    clear @a[tag=buyer.netherite,scores={diamonds=5..}] diamond 5
+    tellraw @a[tag=buyer.netherite,scores={diamonds=5..}] {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+    tellraw @a[tag=buyer.netherite,scores={diamonds=..4}] {"text":"You don't have 5 diamonds","color":"dark_red"}
+    tag @a remove buyer.netherite
+    
+Or if you prefer a function:
+
+    # function example:load
+    scoreboard objectives add diamonds dummy
+    
+    # function example:buy/netherite
+    execute store result score @s diamonds if items entity @s container.* diamond
+    execute if entity @s[scores={diamonds=5..}] run function example:buy/netherite/success
+    execute unless entity @s[scores={diamonds=5..}] run tellraw @s {"text":"You don't have 5 diamonds","color":"dark_red"}]
+    
+    # function example:buy/netherite/success
+    give @s netherite_ingot 1
+    clear @s diamond 5
+    tellraw @s {"text":"You bought a netherite ingot for 5 diamonds","color":"green"}
+
+> [!NOTE]
+> The function `example:buy/netherite` must be run `as` the player
+
+### Bedrock
+In bedrock we have the `hasitem` argument, so it uses less commands than in Java.
+
+#### With an NPC
```

Comment:
since you're mentioning NPCs here, what about using villagers in Java edition?

Reply:
Like using the vanilla trading sistem or adding the commands with thw talk villager scoreboard criteria?

Reply:
vanilla trading system. don't need to explain it in depth, just mention that it's a possibility, maybe give an example command and link to mcstacker for example.